src/longestCommonPxth.ts

```diff
@@ -0,0 +1,37 @@
+import { createPxth } from './createPxth';
+import { getPxthSegments } from './getPxthSegments';
+import { Pxth } from './Pxth';
+import { PxthSegments } from './PxthSource';
+
+/**
+ * Finds longest common path.
+ * @example
+ * [['hello', 'world'], ['hello', 'world, 'yes'], ['hello', 'world', 'bye', 'asdf']] -> 'hello.world'
+ * [['a'], ['b'], ['c']] -> RootPathToken
+ */
+export const longestCommonPxth = (paths: Pxth<unknown>[]): Pxth<unknown> => {
+    if (paths.length === 0) {
+        return createPxth([]);
+    }
+
+    const segmentedPaths = paths.map(getPxthSegments);
+
+    const resultingSegments: PxthSegments = [];

```

Comment:
No need for separate variable for the result - you can simply call `segmentedPaths[0].slice` to take `i` first segments of the path, when returning the result.
Also, this would have slightly better performance, for the cases, where `segmentedPaths[0]` is the longest common path - you can simply return it, without additional `createPxth` call, or slices.